pkg/rest/client.go

```diff
@@ -102,7 +107,7 @@ func NewDynatraceClient(environmentUrl, token string) (DynatraceClient, error) {
 		return nil, errors.New("environment url " + environmentUrl + " was not valid")
 	}
 
-	if parsedUrl.Scheme != "https" {
+	if parsedUrl.Scheme != "https" && !skipHttpsCheck {
```

Comment:
Tbh not a fan of this, but testing with a TLS server is not easily possible. https://stackoverflow.com/a/54955508

Reply:
Hm... the way I understand the base question (https://stackoverflow.com/q/54899550) this would work for the client produced by the httptest tls server. 

So maybe rather than just added the skip option change newDynatraceClient to take the client as input?
It otherwise just creates a base client on line 122, you can pass that in from the public one, and plug in the correct test client for testing